### PR TITLE
LPD-32759 Technical Task | Create document types via headless API

### PR DESCRIPTION
**Pull request by Mikellorza**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-32759

Create headless apis to create document types in a site or in an assetlibrary. The endpoints have been created using some data engine objects like dataDefinitionFields an dataLayout, and, this is good because we are delegating to them the validation and the creation of some objects.

# How am I fixing it?

Nothing to fix. The PR contains the new functionally to create document types via headless apis.

# How can you verify that it works?

Run the included integrations tests.

### CODE REVIEW

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentDataDefinitionTypeDTOConverter.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		return DocumentType.class.getSimpleName();
+	}
+
+	@Override
+	public DocumentDataDefinitionType toDTO(
+			DTOConverterContext dtoConverterContext,
+			DLFileEntryType dlFileEntryType)
+		throws Exception {
+
+		Group group = _groupLocalService.fetchGroup(
```
Comment (jkappler) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentDataDefinitionTypeDTOConverter.java:51]: @Mikellorza, just a minor change. Fetching here can throw an NPE in GroupUtil.getAssetLibraryKey or  GroupUtil.getSiteId(group)

**Review by jkappler — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentDataDefinitionTypeDTOConverter.java (hunk 1 above)
Comment (Mikellorza) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/converter/DocumentDataDefinitionTypeDTOConverter.java]: @jkappler I've changed it to getGroup.

thanks for the revision :)

**Comment by jkappler** [comment]
ci:forward

**Review by jkappler — APPROVED** [review]

**Comment by Mikellorza** [comment]
ci:report:80629945

**Comment by Mikellorza** [comment]
ci:report:80629945

**Comment by Mikellorza** [comment]
There are failing some tests that make no sense, so, let's run test:relevant suite again in order to see if tests are still failing.

**Comment by Mikellorza** [comment]
ci:test:relevant

**Comment by Mikellorza** [comment]
ci:report:82527149

**Comment by Mikellorza** [comment]
Analyzing the last run of the relevant I can say that there are not related failures with the changes of this PR. So, let's move forward to Brian's repo.

**Comment by Mikellorza** [comment]
ci:forward:force

**Pull request closed without merging**